ldclient.go

```diff
@@ -678,7 +732,40 @@ func (client *LDClient) BoolVariationDetail(
 	context ldcontext.Context,
 	defaultVal bool,
 ) (bool, ldreason.EvaluationDetail, error) {
-	detail, err := client.variation(key, context, ldvalue.Bool(defaultVal), true, client.eventsWithReasons)
+	detail, err := client.variationWithHooks(gocontext.TODO(), key, context, ldvalue.Bool(defaultVal), true,
+		client.eventsWithReasons, boolVarDetailFuncName)
+	return detail.Value.BoolValue(), detail, err
+}
+
+// BoolVariationEx is the same as [LDClient.BoolVariation], but accepts a context.Context.
+//
+// Returns defaultVal if there is an error, if the flag doesn't exist, or the feature is turned off and
+// has no off variation.
+//
+// For more information, see the Reference Guide: https://docs.launchdarkly.com/sdk/features/evaluating#go
+func (client *LDClient) BoolVariationEx(
```

Comment:
If the `Ex` methods are only differentiated by taking a Context, would you consider `BoolVariationCtx` instead as the convention?

Reply:
I considered `Context` and `Ctx` which I think make sense if your original function didn't also take a thing called `Context`. When I started calling them it seemed a little strange.

We could also add functional options right now as well, which would make them different, but I cannot think of any options to actually add. So they would be placeholder.

I could change to Ctx if you prefer.

Reply:
To my Go-brain, Ctx seems intuitive and I would prefer it. If I was unfamiliar with the LD concept of context I'd immediately know this took a `context.Context`.

Reply:
Side note: I think it'd be good to explain what the context is used for here. Like, the programmer might think "Ok, this takes a context, it must be a long running operation. Therefore if I want to end evaluation early, I should call context.Cancel()`.

Reply:
I did make a try at this, then I wondered if it would be a problem. If we decide we want to use the context for something, will it become a problem. 

Maybe it wouldn't be a problem to specifically say that cancelling a context does not cancel an evaluation. Versus saying specifically what we use it for.